nixos/modules/services/x11/redshift.nix

```diff
@@ -106,33 +78,20 @@ in {
   };
 
   config = mkIf cfg.enable {
-    assertions = [ 
-      {
-        assertion = 
-          if cfg.provider == "manual"
-          then (cfg.latitude != null && cfg.longitude != null) 
-          else (cfg.latitude == null && cfg.longitude == null);
-        message = "Latitude and longitude must be provided together, and with provider set to null.";
-      }
-    ];
-
     # needed so that .desktop files are installed, which geoclue cares about
     environment.systemPackages = [ cfg.package ];
 
-    services.geoclue2 = mkIf (cfg.provider == "geoclue2") {
-      enable = true;
-      appConfig."redshift" = {
-        isAllowed = true;
-        isSystem = true;
-      };
+    services.geoclue2.appConfig."redshift" = {
+      isAllowed = true;
+      isSystem = true;
     };
 
-    systemd.user.services.redshift = 
+    systemd.user.services.redshift =
     let
-      providerString = 
-        if cfg.provider == "manual"
-        then "${cfg.latitude}:${cfg.longitude}"
-        else cfg.provider;
+      providerString =
+        if lcfg.latitude != null && lcfg.longitude != null
+        then "${lcfg.latitude}:${lcfg.longitude}"
+        else lcfg.provider;
```

Comment:
I think this should be
```
{
  providerString =
    if lcfg.provider == "manual"
    then "${lcfg.latitude}:${lcfg.longitude}"
    else lcfg.provider;
}
```

NixOS takes care of informing the user of unset options, and they do need to set `lat` and `lon` if they want to use the `manual` provider.